gilmour.go

```diff
@@ -523,16 +536,14 @@ func isExpired(timeout time.Duration, startTime time.Time) bool {
 	return time.Now().After(startTime.Add(timeout))
 }
 
-func try(r RetryConf, fn func() error) (err error) {
+func try(g *Gilmour, fn func() error) (err error) {
 	startTime := time.Now()
 	for {
 		err = fn()
-		if isExpired(r.Timeout, startTime) || !isNetworkError(err) || err == nil {
+		if isExpired(g.getRetryTimeout(), startTime) || !isNetworkError(err) || err == nil {
```

Comment:
@meson10 timeout makes sense because in case if the user is using redis sentinel as backend, that timeout is based on failover time of sentinel. Should the user be expected to calculate number of retries and feed it to gilmour?